src/main/scala/strawman/collection/View.scala

```diff
@@ -290,22 +287,107 @@ trait ArrayLike[+A] extends Any {
   def apply(i: Int): A
 }
 
+trait SeqView[+A] extends View[A] with SeqOps[A, SeqView, SeqView[A]] { self =>
+  override def view: SeqView[A] = this
+
+  override def toSeq = immutable.IndexedSeq.from(this)
+
+  override def iterableFactory: IterableFactoryLike[SeqView] = SeqView
+
+  override protected[this] def fromSpecificIterable(coll: Iterable[A]): SeqView[A] = fromIterable(coll)
+
+  override protected[this] def newSpecificBuilder(): Builder[A, SeqView[A]] =
+    IndexedSeq.newBuilder().mapResult(_.view)
+
+  override def toString = "SeqView(?)"
+  override def className = "SeqView"
+
+  override def apply(i: Int) = iterator().drop(i - 1).next()
+  override def prepended[B >: A](elem: B): SeqView[B] = SeqView.Prepend(elem, this)
+  override def appended[B >: A](elem: B): SeqView[B] = SeqView.Append(this, elem)
+  override def prependedAll[B >: A](prefix: Iterable[B]): SeqView[B] = SeqView.PrependAll(prefix, this)
+  override def appendedAll[B >: A](suffix: Iterable[B]): SeqView[B] = SeqView.AppendAll(this, suffix)
+}
+
+object SeqView extends SeqFactory[SeqView] {
```

Comment:
Unfortunately in `SeqFactory` the `Source` type is fixed to `IterableOnce`, which is incompatible with views. We should extend `IterableFactoryLike` instead.

I’m wondering if we should make the `Source[A]` type member in `IterableFactoryLike` not lower bounded so that we could fix it to `Seq[A]` here. But we could try that in a different PR.